lib/streamlit/hashing.py

```diff
@@ -375,21 +384,17 @@ def _to_bytes(self, obj, context):
                 self._update(h, obj.keywords)
                 return h.digest()
             else:
-                try:
-                    # As a last resort, we pickle the object to hash it.
-                    return pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)
-                except:
-                    st.warning(
-                        _hashing_error_message(
-                            "Streamlit cannot hash an object of type %s." % type(obj)
-                        )
-                    )
-        except:
-            st.warning(
-                _hashing_error_message(
-                    "Streamlit failed to hash an object of type %s." % type(obj)
-                )
-            )
+                # As a last resort
+                h = hashlib.new(self.name)
+                # add type to distingush x from [x]
```

Comment:
I don't really understand this comment.

Reply:
This is so that `["a"]` hashes differently from `("a")` and `"a"`

We hash the name of the container along with the elements to differentiate the data. Otherwise we'd just be hashing the data and the hashes would be the same.

Should I modify the comment? @tvst

Reply:
I also find this comment not as explanatory as it could be.  You could write more, IMHO.

AFAIC we could use more comments in the caching/hashing code.  It's a confusing place!